src/System.Private.CoreLib/shared/System/Globalization/DateTimeFormatInfo.cs

```diff
@@ -266,6 +267,8 @@ internal DateTimeFormatInfo(CultureData cultureData, Calendar cal)
             _cultureData = cultureData;
 
             Calendar = cal;
+            // TODO-NULLABLE: workaround compiler issue
+            calendar = calendar!;
```

Comment:
Will the C# compiler output this in the generated IL?  I believe it will, in which case we should not be making changes like this just to suppress the warnings.

Reply:
@cston just provided a good solution here, which is initialize the field with `null!` rather than doing this. 

https://github.com/dotnet/coreclr/pull/23560#discussion_r270510854